DAT-699 make highlighting work with advanced search & DAT-774 highlighting improvements

hareshkainthtpx opened the pull request:
The method was switched from "unified" to "original" due to an issue with highlighting matched values in titles. While "unified" is more efficient for large documents, resolving the title highlight issue is currently prioritized.

Commit:
fix:change highlight method to "original" in search params

The method was switched from "unified" to "original" due to an issue with highlighting matched values in titles. While "unified" is more efficient for large documents, resolving the title highlight issue is currently prioritized.

RickMoynihan commented:
Thanks @hareshkainthpixel, I'll have a play with the PR as we experimented a little with other `hl.methods` before setting this as it seemed to have slightly better behaviour in some cases.

I'd like to know why this change fixes it, as it looked like from your debug statements that you showed me whilst pairing that solr was doing the right thing on the title field.

RickMoynihan committed:
Fix enabling advanced boolean search to work with highlighting

Unlike the attempt in the previous commit this doesn't change the
highlighter implementation, which means we can still configure it
through the other options on the request.

RickMoynihan committed:
Tweaks to highlighting to improve behaviour regarding DAT-774

https://london.atlassian.net/browse/DAT-774

We also extend the set of VALID_SOLR_PARAMETERS to make
future experimentation easier with these values.

Acked-by: Rick Moynihan <rick.m@swirrl.com>

RickMoynihan commented:
I have updated this PR not to change the highlight method, but to also work with the existing unified highlighter, which has options we want to use to improve with regards to DAT-774 (giving users better context on search descriptions).
